File: app/src/main/java/io/eodc/planit/model/SubjectViewModel.java
package io.eodc.planit.model;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.support.annotation.NonNull;

import io.eodc.planit.db.PlannerDatabase;
import io.eodc.planit.db.Subject;
import io.eodc.planit.db.SubjectDao;

public class SubjectViewModel extends AndroidViewModel {
    private Subject mSubject;

    public SubjectViewModel(@NonNull Application application, int id) {
        super(application);
        SubjectDao subjectDao = PlannerDatabase.getInstance(application).classDao();
        mSubject = subjectDao.getClassById(id);
    }

    public Subject getSubject() {
        return mSubject;
    }
}

File: app/src/main/java/io/eodc/planit/model/SubjectViewModel.kt
package io.eodc.planit.model

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import io.eodc.planit.db.PlannerDatabase
import io.eodc.planit.db.Subject

class SubjectViewModel(application: Application, id: Int) : AndroidViewModel(application) {
    val subject: Subject

    init {
        val subjectDao = PlannerDatabase.getInstance(application)!!.classDao()
        subject = subjectDao.getClassById(id)
    }
}
